tests/system/test_universe_domain.py

```diff
@@ -0,0 +1,94 @@
+import pytest
+
+import grpc
+
+# We are not able to stack multiple dimensions of
+# `@pytest.mark.parametrize` for multiple parameters in the tests that
+# follow, so we flatten each combination of parameter values into a
+# single EchoParams object with multiple attributes, and parametrize
+# the test only on EchoParams.
+
+
+class EchoParams(object):
+    def setattrs(self, **kwargs):
+        for k, v in kwargs.items():
+            setattr(self, k, v)
+        return self
+
+# What we'd like for the following test is:
+#    @pytest.mark.parametrize("channel_creator", [grpc.insecure_channel])
+#    @pytest.mark.parametrize("transport_name",["grpc","rest"])
+
+
+def vary_transport():
+    for transport_name in ["grpc", "rest"]:
+        params = EchoParams()
+        params.setattrs(
+            channel_creator=grpc.insecure_channel,
+            transport_name=transport_name,
+            transport_endpoint="localhost:7469",
+            credential_universe="googleapis.com",
+            client_universe="googleapis.com"
+        )
+        yield params
+
+
+@pytest.mark.parametrize("test_params", vary_transport())
+def test_universe_domain_validation_pass(parametrized_echo, test_params):
+    # Test that only the configured client universe and credentials universe are used for validation
+    assert parametrized_echo.universe_domain == test_params.client_universe
+    assert parametrized_echo.transport._credentials._universe_domain == test_params.credential_universe
+    if test_params.transport_name == "rest":
+        assert parametrized_echo.api_endpoint == "http://" + test_params.transport_endpoint
+    else:
+        assert parametrized_echo.api_endpoint == test_params.transport_endpoint
+    response = parametrized_echo.echo({
+        'content': 'Universe validation succeeded!'
+        })
+    assert response.content == "Universe validation succeeded!"
+
+
+# What we'd like to do for the following test is:
+#    @pytest.mark.parametrize("channel_creator", [grpc.insecure_channel])
+#    @pytest.mark.parametrize("transport_name",["grpc","rest"])
+#    @pytest.mark.parametrize("transport_endpoint", ["showcase.googleapis.com", "localhost:7469" ])
+#    @pytest.mark.parametrize("credential_universe",["showcase.googleapis.com", "localhost:7469"])
```

Comment:
Remove commented code. Add a comment that reads like a comment instead of code. Applies throughout.

Reply:
Drive-through comment: the comment would be about not getting this code to work, so including the code seems the most concise way. Maybe we can expand the preamble, saying the this attempted parametrization stacking was not working.

Reply:
@ohmayr , Please could you take a look to see if we can achieve the same result using `@pytest.mark.parametrize` similar to how we do it here? Let me know if you'd like to pair program on it

https://github.com/googleapis/gapic-generator-python/blob/2e86a654896054f1ac4b6833127bb095d2bf5b90/tests/integration/goldens/asset/tests/unit/gapic/asset_v1/test_asset_service.py#L223-L227